### PR TITLE
style: Support ::slotted better.

### PR DESCRIPTION
**Pull request by emilio**
This allows to selector-match ::slotted, though we still don't parse it.

Bug: 1425834, 1424607, 1425755
Reviewed-by: heycam
MozReview-Commit-ID: ItELHkf2PMl

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19721)

### CODE REVIEW

**Comment by emilio** [comment]
@bors-servo retry

 * Gecko-media build fail

**Comment by tigercosmos** [comment]
```
error: no rules expected the token `"slot"`
   --> components/style/dom.rs:424:48
    |
424 |         self.get_local_name() == &*local_name!("slot") &&
    |                                                ^^^^^^
```

**Comment by CYBAI** [comment]
@tigercosmos No worries, I think it will be fixed soon lol

**Comment by nox** [comment]
I published markup5ever 0.7.1, now we just need to bump it in this PR.

**Comment by jdm** [comment]
```
error[E0046]: not all trait items implemented, missing: `is_html_element`
   --> components/layout_thread/dom_wrapper.rs:352:1
    |
352 | / impl<'le> TElement for ServoLayoutElement<'le> {
353 | |     type ConcreteNode = ServoLayoutNode<'le>;
354 | |     type TraversalChildrenIterator = DomChildren<Self::ConcreteNode>;
355 | |
...   |
534 | |     }
535 | | }
    | |_^ missing `is_html_element` in implementation
    |
    = note: `is_html_element` from trait: `fn(&Self) -> bool`

error: aborting due to previous error

error: Could not compile `layout_thread`.

```

**Comment by jdm** [comment]
@bors-servo retry
Let's see if it was an isolated problem.

File: ports/glutin/Cargo.toml [hunk 1]
```diff
@@ -42,5 +42,5 @@ gdi32-sys = "0.2"
 [target.'cfg(target_os = "macos")'.dependencies]
 osmesa-src = {git = "https://github.com/servo/osmesa-src"}
 
-[target.x86_64-unknown-linux-gnu.dependencies]
-osmesa-src = {git = "https://github.com/servo/osmesa-src"}
+# [target.x86_64-unknown-linux-gnu.dependencies]
+# osmesa-src = {git = "https://github.com/servo/osmesa-src"}
```
Comment (jdm) [inline ports/glutin/Cargo.toml]: This change is causing the errors with loading OSMesa during testing.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by emilio** [comment]
Welp, I should really really update osmesa :)

@bors-servo r=heycam

**Comment by emilio** [comment]
@bors-servo retry

 * Timeouts look flaky.

**Pull request merged**